easy-ui-react/src/TextField/TextField.tsx

```diff
@@ -0,0 +1,359 @@
+ * import { TextField } from "@easypost/easy-ui/TextField";
+ *
+ * export function Component() {
+ *  const [email, setEmail] = useState("");
+ *  return (
+ *    <>
+ *      <TextField
+ *        type="email"
+ *        label="Email address"
+ *        autoFocus
+ *        value={email}
+ *        onChange={(inputValue) => setEmail(inputValue)} // value is returned automatically via react-aria
+ *        isRequired
+ *      />
+ *      <span>email entered: {email}</span>
+ *    </>
+ *  );
+ * }
+ * ```
+ * @example
+ * _Password variation with helper text:_
+ * ```tsx
+ * import { TextField } from "@easypost/easy-ui/TextField";
+ *
+ * export function Component() {
+ *  const [password, setPassword] = useState("");
+ *  return (
+ *    <>
+ *      <TextField
+ *        type="password"
+ *        label="Password"
+ *        helperText={<a href={RESET_URL}>Forgot password? </a>}
+ *        value={password}
+ *        onChange={(inputValue) => setPassword(inputValue)} // value is returned automatically via react-aria
+ *        isRequired
+ *      />
+ *    </>
+ *  );
+ * }
+ * ```
+ * @example
+ * _Visually hidden label with left aligned icon and placeholder text:_
+ * ```tsx
+ * import { TextField } from "@easypost/easy-ui/TextField";
+ *
+ * export function Component() {
+ *  const [searchedValue, setSearchedValue] = useState("");
+ *  return (
+ *    <>
+ *      <TextField
+ *        type="search"
+ *        label="Search for carriers" // visually hidden but still accessible via isLabelVisuallyHidden prop
+ *        isLabelVisuallyHidden
+ *        iconAtStart={SearchIcon}
+ *        value={searchedValue}
+ *        onChange={(inputValue) => setSearchedValue(inputValue)} // value is returned automatically via react-aria
+ *        placeholder="FedEx, UPS, USPS"
+ *        isRequired
+ *      />
+ *    </>
+ *  );
+ * }
+ * ```
+ * @example
+ * _Invalid state with error text:_
+ * ```tsx
+ * import { TextField } from "@easypost/easy-ui/TextField";
+ *
+ * export function Component() {
+ *  const [value, setValue] = useState("");
+ *  return (
+ *    <>
+ *      <TextField
+ *        label="Error"
+ *        validationState="invalid"
+ *        helperText="Some text" // will be overriden in the presence of an invalid state with error text
+ *        errorText="Some error text"
+ *        value={value}
+ *        onChange={(inputValue) => setValue(inputValue)} // value is returned automatically via react-aria
+ *        isRequired
+ *      />
+ *    </>
+ *  );
+ * }
+ * ```
+ */
+export function TextField(props: TextFieldProps) {
+  const {
+    type = "text",
+    size = "md",
+    isLabelVisuallyHidden = false,
+    isDisabled = false,
+    isRequired = false,
+    validationState = "valid",
+    emphasizedLabel = false,
+    autoFocus = false,
+    label,
+    errorText,
+    helperText,
+    placeholder,
+    value,
+    defaultValue,
+    iconAtStart,
+    iconAtEnd,
+  } = props;
+  const [isVisible, setIsVisible] = useState(false);
```

Comment:
**nit:** the name `isVisible` wasn't immediately obvious to me what it represented since we have other concepts around visibility (like label being visually hidden). it's not a big deal if there isn't a better name, it may be fine as-is, but wondering if something else would be more clear like `isPasswordVisible` or `isInputValueVisible`

Reply:
updated to `isPasswordVisible`